apps/store/src/services/cart/useCart.ts

```diff
@@ -0,0 +1,25 @@
+import { useState } from 'react'
+import { useCurrentLocale } from '@/lib/l10n/useCurrentLocale'
+import { useShopSessionFindOrCreateQuery } from '@/services/apollo/generated'
+import { shopSessionServiceInitClientSide } from '@/services/shopSession/ShopSessionService'
+
+const shopSessionService = shopSessionServiceInitClientSide()
+
+export const useCart = () => {
+  const { countryCode } = useCurrentLocale()
+  const [shopSessionId, setShopSessionId] = useState(() => {
+    const shopSessionId = shopSessionService.shopSessionId
+    return shopSessionId
+  })
+
+  const result = useShopSessionFindOrCreateQuery({
+    variables: { shopSessionId, countryCode },
+    onCompleted(data) {
+      if (data.shopSessionFindOrCreate.id !== shopSessionId) {
+        setShopSessionId(data.shopSessionFindOrCreate.id)
+      }
+    },
+  })
+
+  return result
```

Comment:
Since it's called `useCart` could we change it to:

```suggestion
  return result.data?.shopSessionFindOrCreate.cart
```

Reply:
hmmm I guess but then we loose the additional data like if the query is loading 🤔

But yeah, I agree that we should parse the response somewhat.

Reply:
True 🤔 
Maybe we could return something like `[cart, { loading, errors }]`